File: usr/openData-SqlDrive/SqlDrive/Node.JS/SqlDrive.js

var _md5 = require("./Md5");
function setRemote(remote) {
    /**
     * Remote Value = Url + Port || Or only use url;
     */
    this.remote = remote;
}
function setLoginUser(user) {
    this.user = user;
}
function setLoginPasswd(passwd){
    this.passwd = _md5._md5(passwd);
}
function run(command) {
    //console.log(this.user+" "+this.passwd+" "+this.command);
    var http = require('http');
    try{
        http.get(this.remote+"/?Logon="+this.user+"?Passwd="+this.passwd+"?Command="+command, function(res) {
            var data = '';
            var mes = "";
            res.setEncoding("utf8");
            res.on('data', function(chunk) {
                data += chunk;
            }).on('end', function() {
                var s = data.split("\n");
                for (var i = 0 ; i < s.length ; i++) {
                    if (s[i] == "") {
                        break;
                    }
                    console.log(" -- "+s[i]);
                }
            }).on('err', function () {
                return false;
            });
        });
        return true;
    }catch (err) {
        console.log(err);
        return false;
    }
}
function createData(name,value,note,database) {
    run("create data '"+name+"' setting('"+value+"','"+note+"') in "+database);
}
function createDatabase(name) {
    run("create database '"+name+"'");
}
function deleteData(name,database) {
    run("delete data '"+name+"' in "+database);
}
function deleteDatabase(name) {
    run("delete database "+name);
}
function findData(name) {
    run("find data "+name);
}
function findDatabase(name) {
    run("find database "+name);
}
function indexData(name,database) {
    run("index '"+name+"' in "+database);
}
function reNameDatabase(database,rename) {
    run("rename database '"+database+"' '"+rename+"'");
}
function reNameData(data,rename,database) {
    run("rename data '"+database+"' '"+rename+"' in "+database);
}

module.exports = {
    createData,
    createDatabase,
    deleteData,
    deleteDatabase,
    findData,
    findDatabase,
    indexData,
    reNameData,
    reNameDatabase,
    setLoginPasswd,
    setLoginUser,
    setRemote
};

File: usr/openData-SqlDrive/SqlDrive/Python/SqlDrive/SqlDrive.py
import requests
import hashlib

class SQL_DO(object):
    def __init__(self,remote,user,passwd) -> None:
        self.userName = user
        self.Passwd = passwd
        self.Remote = remote

    def add_MD5(self,addStr:str)->str:
        hl = hashlib.md5()
        hl.update(addStr.encode("utf-8"))
        return hl.hexdigest()

    def sendMysScript(self,script:str) -> bool:
        #print(str(self.Remote)+"/?Logon="+str(self.userName)+"?Passwd="+str(self.add_MD5(self.Passwd))+"?Command="+script)
        r = requests.get(str(self.Remote)+"/?Logon="+str(self.userName)+"?Passwd="+str(self.add_MD5(self.Passwd))+"?Command="+script)
        getText = r.text
        split = getText.split("\n")
        message:str = ""
        j = 0
        for i in split:
            j = j + 1
            message = message + str(i) + "\n"
        
        j = j - 1
        self.Message = message
        if j == 0:
            return False
        else:
            return True
    def getMessage(self) -> str:
        return self.Message
    def outValue(self) -> None:
        print(self.userName+" "+self.Passwd+" "+self.Remote)

    def listDatabase(self) -> list[str]:
        if self.sendMysScript("list database") == True:
            return self.getMessage().split("\n")
        else:
            return []
    
    def deleteDatabase(self,name:str) -> bool:
        return self.sendMysScript("delete database "+str(name))
    
    def deleteData(self,name:str,database:str) -> bool:
        return self.sendMysScript("delete data '"+str(name)+"' in "+database)

    
